test/Autofac.Integration.AspNetCore.Multitenant.Test/AutofacMultitenantServiceProviderExtensionsTests.cs

```diff
@@ -11,7 +12,9 @@ public sealed class AutofacMultitenantServiceProviderExtensionsTests
         [Fact]
         public void GetAutofacMultitenantRootReturnsMultitenantContainer()
         {
-            var serviceProvider = new AutofacMultitenantServiceProvider(CreateMultitenantContainer());
```

Comment:
I'm more concerned that you can't "new" one of the providers up (ie, it requires the magic of the factory) than I am that the middleware is required. Any way we can keep the provider working without factory magic?

Reply:
That's what I was trying to explain earlier in one of the comments. That's why I'd make that provider internal or maybe even remove it and only use `AutofacServiceProvider` and pass in the MTC. What do you think about it? I couldn't think about another solution here.

Reply:
I see what you're saying, but, like... it seems by tying the class so hard to the factory it's getting a little spaghetti.

- The provider by itself doesn't behave 100% correctly without it being created by the factory.
- The part that doesn't behave right is an _extension method_ on the class, not even something right on the class.

So we end up with this sort of indirect tie between three classes that are somewhat intended to be separate items. I think I need to understand more before I can suggest something else.